tests/src/test/java/org/eclipse/hono/tests/amqp/AmqpAdapterBase.java

```diff
@@ -1,289 +1,151 @@
-
-        final Async addTenantTracker = context.async();
-        final TenantObject tenant = TenantObject.from(tenantId, true);
-        helper.registry.addDeviceForTenant(tenant, deviceId, password)
-                .setHandler(context.asyncAssertSuccess(ok -> addTenantTracker.complete()));
-        addTenantTracker.await();
-
-        final Async setup = context.async();
-
-        final HonoClient client = prepareAmqpAdapterClient(username, password);
-        client.connect(new ProtonClientOptions()).compose(ok -> {
-            clientsToShutdown.add(client);
-            return Future.succeededFuture();
-        }).compose(ok -> createConsumer(tenantId, msg -> {
-            received.countDown();
-            if (received.getCount() % 40 == 0) {
-                LOG.info("message {} received", MESSAGES_TO_SEND - received.getCount());
-            }
-        })).setHandler(context.asyncAssertSuccess(ok -> setup.complete()));
-        setup.await();
+    @Override
+    protected HonoClient prepareHonoClient(final TestContext ctx) {
+        final String username = IntegrationTestSupport.getUsername(AMQP_ADPATER_TEST_DEVICE, AMQP_ADAPTER_TEST_TENANT);
 
-        final AtomicInteger messageCount = new AtomicInteger(0);
-
-        while (messageCount.get() < MESSAGES_TO_SEND) {
-            final Async sendingComplete = context.async();
-            getOrCreateSender(tenantId, client).compose(sender -> {
-                final Message message = ProtonHelper.message("temp: " + messageCount.getAndIncrement());
-                sender.send(message).setHandler(sendOutcome -> {
-                    if (sendOutcome.succeeded()) {
-                        if (messageCount.get() % 40 == 0) {
-                            LOG.info("messages sent: {}", messageCount.get());
-                        }
-                    } else {
-                        LOG.debug("error sending message {}", messageCount.get(), sendOutcome.cause());
-                    }
-                    sendingComplete.complete();
-                });
-
-                return Future.succeededFuture();
-            });
-            sendingComplete.await();
-        }
-
-        final long timeToWait = Math.max(TEST_TIMEOUT - 1000, Math.round(MESSAGES_TO_SEND * 20));
-        received.await(timeToWait, TimeUnit.MILLISECONDS);
-        final long messagesReceived = MESSAGES_TO_SEND - received.getCount();
-        LOG.info("sent {} and received {} messages", messageCount, messagesReceived);
-        if (messagesReceived < messageCount.get()) {
-            context.fail("did not receive all messages");
-        }
-    }
-
-    // ------------------------------< private methods >---
-
-    private static HonoClient prepareAmqpAdapterClient(final String username, final String password) {
         final ClientConfigProperties config = new ClientConfigProperties();
-        config.setName("test");
+        config.setName("EventAmqpIT");
```

Comment:
since this is a generic class, does it make sense to use a name here that suggests we are sending events?

Reply:
not at all :+1: